tools/check_SOILWAT2.sh

```diff
@@ -66,54 +79,145 @@ for ((k = 0; k < ncomp; k++)); do
     fi
   fi
 
-  echo $'\n'`"${ccs[k]}" --version`
-  echo $'\n'`"${cxxs[k]}" --version`
-  echo $'\n'\
-       -----------------------------------------------------
-
   CC=${ccs[k]} CXX=${cxxs[k]} make compiler_version
 
 
   echo $'\n'$'\n'\
        --------------------------------------------------$'\n'\
-       Run binary with ${port_compilers[k]} ...$'\n'\
+       ${k}'-a)' Run example simulation with ${port_compilers[k]} ...$'\n'\
        --------------------------------------------------
 
-  CC=${ccs[k]} CXX=${cxxs[k]} make clean bin_run
+  echo $'\n'"Target 'bin_run' ..."
+  res=$(CC=${ccs[k]} CXX=${cxxs[k]} make clean bin_run 2>&1)
+
+  res_error=$(check_error "${res}")
+  if [ "${res_error}" ]; then
+    echo "Failure:"
+    if [ "${verbosity_on_error}" = true ]; then
+      echo "${res}"
+    else
+      echo "${res_error}"
+    fi
+
+  else
+    echo "Success: target."
 
-  if [ ${k} -eq 0 ]; then
-    if [ ! -d "${dir_out_ref}" ]; then
-      echo "Save default testing output as reference for future comparisons"
-      cp -r tests/example/Output "${dir_out_ref}"
+    if [ ${k} -eq 0 ]; then
+      if [ ! -d "${dir_out_ref}" ]; then
+        echo "Save default testing output as reference for future comparisons"
+        cp -r tests/example/Output "${dir_out_ref}"
+      fi
     fi
+
+    check_testing_output
+  fi
+
+  echo $'\n'"Target 'bin_debug_severe' ..."
+  res=$(CC=${ccs[k]} CXX=${cxxs[k]} make clean bin_debug_severe 2>&1)
+
+  res_error=$(check_error "${res}")
+  if [ "${res_error}" ]; then
+    echo "Failure:"
+    if [ "${verbosity_on_error}" = true ]; then
+      echo "${res}"
+    else
+      echo "${res_error}"
+    fi
+
+  else
+    echo "Success: target."
```

Comment:
This is nit-picky, but I have a suggestion:

Wouldn't it make it look more uniform  if successes/failures were along the lines of:
```
Result: Success
Result: Failed
```
and extend that to the other successes/failure messages? Is there a reason against this?

Reply:
I see your point and think it works well for the targets that run the test cases. However, how would you suggest to update the output of `compare_output_against_reference()` to fit this pattern?

Reply:
I see your concern with `compare_output_against_reference()`. So my suggestion is something along the lines of:

```
"Output Result: Success - Output reproduces reference exactly."
"Output Result: Failed - Output deviates beyond tolerance from reference:"
"Output Result: Success - Output reproduces reference within tolerance but not exactly:"
```
which is a bit cluttered, but does fit the pattern.